### PR TITLE
Extend admin-client GroupsResource: Enhanced query functionality

### LINKED ISSUE #20016: Group attribute query is missing QueryParams in java admin client
### Description

Using the Java admin client library it is currently not possible to query attributes with additional parameters. Thus, the parameter described in the REST API  (first, max, and briefRepresentation) cannot be set when using the admin-client library.

### Discussion

_No response_

### Motivation

The admin client library should match the REST API documentation.

### Details

#12964 implemented the group attribute search. However the GroupsResource is misssing the additional method with all QueryParams. For reference: [the UsersResource ](https://github.com/keycloak/keycloak/blob/9186327807477f4ca5c36d0adb6fd7b7f62b9721/integration/admin-client/src/main/java/org/keycloak/admin/client/resource/UsersResource.java#L120) implements all parameters.

### PR DESCRIPTION
**Pull request by danielFesenmeyer**
Support the query functionality to be used in combination with the parameters first, max and briefRepresentation

Closes #20016

### CODE REVIEW

**Comment by danielFesenmeyer** [comment]
Rebased, hoping that a flaky test has been fixed in the mean time.

File: integration/admin-client-jee/src/main/java/org/keycloak/admin/client/resource/GroupsResource.java [hunk 1]
```diff
@@ -150,4 +150,11 @@ List<GroupRepresentation> groups(@QueryParam("search") String search,
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     List<GroupRepresentation> query(@QueryParam("q") String searchQuery, @QueryParam("populateHierarchy") boolean populateHierarchy);
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    List<GroupRepresentation> query(@QueryParam("q") String searchQuery,
```
Comment (pedroigor) [inline integration/admin-client-jee/src/main/java/org/keycloak/admin/client/resource/GroupsResource.java:156]: Doesn't make sense to also include `search` and `exact` parameters?

Comment (pedroigor) [reply integration/admin-client-jee/src/main/java/org/keycloak/admin/client/resource/GroupsResource.java:156]: I was more thinking about providing a method that you can use both as we are missing it. But not hard on that.

**Review by pedroigor — APPROVED** [review]

**Comment by pedroigor** [comment]
@danielFesenmeyer Thanks!

**Comment by danielFesenmeyer** [comment]
@pedroigor Thanks for your review!

**Pull request merged**